site/src/Librecores/ProjectRepoBundle/Util/DefaultExecutor.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Librecores\ProjectRepoBundle\Util;
+
+use Psr\Log\LoggerInterface;
+
+use Symfony\Component\Process\ProcessBuilder;
+
+/**
+ * Default implementation of ExecutorInterface
+ *
+ * Uses Symfony ProcessBuilder component to build the process.
+ *
+ * @author Amitosh Swain Mahapatra <amitosh.swain@gmail.com>
+ */
+class DefaultExecutor implements ExecutorInterface
```

Comment:
Lambda based executors ? This also makes testing easier. Using ProcessBuilder made it impossible to test just the algorithm by faking output.